src/generator/templates/cli/lib/engine.mako

```diff
@@ -227,10 +229,24 @@ for parg in ${opt_values(VALUE_ARG)} {
 % for p in optional_props:
 <%
     ptype = actual_json_type(p.name, p.get("format", p.type))
-    default_value = JSON_TO_RUST_DEFAULT[ptype]
-    value_unwrap = f"value.unwrap_or({default_value})"
+    no_default_enum = False
+    if is_property_enum(p):
+        default_value = 'Default::default()'
+        value_unwrap = f"&value.and_then(|v|v.try_into().ok()).unwrap_or({default_value})"
+        no_default_enum = True
+    else:
+        default_value = JSON_TO_RUST_DEFAULT[ptype]
+        value_unwrap = f"value.unwrap_or({default_value})"
 %>\
         "${mangle_subcommand(p.name)}" => {
+        % if no_default_enum:
+            if let Some(value) = value.and_then(|v|v.try_into().ok()) {
+                call = call.${mangle_ident(setter_fn_name(p))}(&value);
+            } else {
+                /*value was empty but has no default*/
+                todo!("I don't know what would be best in this situation.")
```

Comment:
In this case the value provided was either empty or invalid and I'm not sure what to do in this case. 
Should we just ignore this value and move on and see if it throws an error when executing the call?
Maybe output a warning?

Reply:
If it's possible to make the call anyway, the CLI could probably try it and maybe even emit a warning like you suggested.
Admittedly I don't fully understand this yet, but trying to be lenient for the CLI seems like the right choice here.